### ISSUE TITLE
Section 5.1.1

### ISSUE THREAD

**Issue by aaronpk** [pubsubhubbub/PubSubHubbub#31]
Suggestion: Change the text in [Section 5.1.1](https://pubsubhubbub.googlecode.com/svn/trunk/pubsubhubbub-core-0.4.html#rfc.section.5.1.1) to say "The topic URL MUST be the one advertised by the publisher during the discovery phase."

[Section 5.1.1](https://pubsubhubbub.googlecode.com/svn/trunk/pubsubhubbub-core-0.4.html#rfc.section.5.1.1) currently says "The topic URL MUST be the one advertised by the publisher in a Self Link Header during the discovery phase." 

However, [section 4](https://pubsubhubbub.googlecode.com/svn/trunk/pubsubhubbub-core-0.4.html#discovery) says that the topic URL can be specified in _either_ the Link header or a tag in the document body, such as an HTML link tag. This change makes the sections consistent.

**Comment by voxpelli** [comment]
Section 4 should also have its `MUST` in `The HTTP [RFC2616] response from the publisher MUST include at least one Link Header` reduced to a `SHOULD` to better indicate this possibility.

Currently it both specifies that you `MUST` use a header and specifies what you `MAY` use if there is no header. But no header means a non-compliant publisher with the current language, so the need for a fallback seems contradictory by the current spec. (I guess the fallback is there to support PuSH 0.3 publishers?)

**Comment by aaronpk** [comment]
Thanks for the notes, @voxpelli 

oh also @tantek asked me to tag him in this

**Comment by julien51** [comment]
@aaronpk I agree with your change. It's clearer and will avoid any confusion.

@voxpelli I understand your comment and I agree that things are a bit blurry when there are no link headers. Basically think compliance with 0.4 MUST mean that HTTP link headers are used.  However, it is possible to _also_ include Links in other parts (such as the HTML `<head>` section, or `<atom:link>` elements for RSS/Atom feeds... And, as you've identified, when there are no headers, I would much rather let implementers fallback to 0.3 (even though that spec does not apply to HTML).

Basically, inline with the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle), I'd rather have a "strict" spec (with the requirement for link headers) and more flexible implementations (which would tolerate missing Link Headers, as long as there are some in the html/rss). Does it make sense?

**Comment by aaronpk** [comment]
I made a PR for this change! ^^

**Comment by julien51** [comment]
Thanks, merged!
